interpreter/cling/lib/Interpreter/Interpreter.cpp

```diff
@@ -125,11 +125,10 @@ namespace cling {
       // we need a transaction.
       PushTransactionRAII pushedT(i);
 
-      m_State.reset(new ClangInternalState(CI.getASTContext(),
-                                           CI.getPreprocessor(),
-                                           CG ? CG->GetModule() : 0,
-                                           CG,
-                                           "aName"));
+      m_State.reset(new ClangInternalState(
+          CI.getASTContext(), CI.getPreprocessor(),
+          CG ? std::shared_ptr<llvm::Module>(CG->GetModule()) : nullptr, CG,
```

Comment:
This looks strange - should the `ClangInternalState` take ownership of the `Module` produced by `CodeGen`? I'd have bet that the `Transaction` should take ownership, and the `ClangInternalState` gets "observer status" by getting passed a `weak_ptr`.